SU-HSDO/suhumsci
# Prevent menu flash during page-load

# READY FOR REVIEW

## Summary
Our menus are expanded by default so users with JavaScript disabled can navigate the site. This was contributing to a distracting flash of expanded menus during the page load. This update prevents the menu flash.

See [STN-289](https://sparkbox.atlassian.net/browse/STN-289), for more details, and [STN-244](https://sparkbox.atlassian.net/browse/STN-244) for a video demonstrating the issue.

## Need Review By (Date)
4/27

## Urgency
Medium

## Steps to Test
1. Run this branch locally
2. Browse to a page with lots of content (like `/courses` or another courses page).
3. Disable JavaScript, reload the page, and verify that the mobile menu loads fully-expanded. This should happen at both mobile and desktop widths.
4. Enable JavaScript, reload the page and confirm that there is no menu flash on mobile or desktop widths.
    1. For the most accurate test, ensure your browser isn't caching the page when you reload.
    2. For extra testing, you can set your dev tools to simulate a low-bandwidth environment.
5. Generally, test menu navigation (via mouse and keyboard) to ensure there are no regressions.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
